### ISSUE TITLE
Maximum call stack size exceeded - Warning caused by no-inferred-empty-object-type

### ISSUE THREAD

**Issue by Merott** [palantir/tslint#2646]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.3.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import { expect } from 'chai';

describe('app', () => {
  it(`should know numbers`, () => {
    expect(1).to.eq(1);
  });
});
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-inferred-empty-object-type": true
  }
}
```

#### Actual behavior

Outputs `Warning: Maximum call stack size exceeded`.

#### Expected behavior

No warning, or better indicator of what the issue may be.

#### Explanation

I have this issue in a larger project, and managed to narrow it down to the `no-inferred-empty-object-type` rule, and created a minimal test project that reproduces the issue: https://github.com/Merott/tslint-max-call-stack-size-issue-demo.

It appears that it has something to do with the import of `expect` from `chai`, but that may be a red herring, as I couldn't understand where the issue came from.

**Comment by JoshuaKGoldberg** [comment]
The exact issue is coming from your `expect` line. I'm seeing this too with a very similar repro. The issue goes away after removing the `expect(1).to.eq(1);`.

Diving into the source, `isEmptyObjectInterface` is recursing infinitely into itself.
